### ISSUE TITLE
Arrows to navigate to next and previous months are not announcing properly

### ISSUE THREAD

**Issue by jacobsimionato** [flutter/flutter#130300]
Copied from internal bug b/236668243:

When the arrow button for next month is tapped, the button will be announcing the current month. We are expecting it to be announced as a button as well otherwise users are unable to know it is tappable. (This issue applies to previous month button as well.)

## Current behavior:
The button is announcing "July 2022".

## Expected behavior:
"Next month August 2022 button."
Or a better behavior would be "Current month is July 2022, next month button".

<details>
  <summary>Minimal repro code</summary>

```
import 'dart:async';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  DateTime currentDate = DateTime.now();

  Future<void> _selectDate(BuildContext context) async {
    final pickedDate = await showDatePicker(
      context: context,
      initialDate: currentDate,
      firstDate: DateTime(2015),
      lastDate: DateTime(2050),
    );
    if (pickedDate != null && pickedDate != currentDate) {
      setState(() {
        currentDate = pickedDate;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("DatePicker"),
      ),
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Text(currentDate.toString()),
            ElevatedButton(
              onPressed: () => _selectDate(context),
              child: const Text('Select date'),
            ),
          ],
        ),
      ),
    );
  }
}
```
</details>

**Comment by dam-ease** [comment]
Seems close to https://github.com/flutter/flutter/issues/96643
